WordPress/WordPressTest/BlogJetpackTests.swift

```diff
@@ -0,0 +1,264 @@
+            syncExpectation.fulfill()
+        } failure: { _ in
+            XCTFail("Sync blogs shouldn't fail")
+        }
+        wait(for: [syncExpectation], timeout: 5.0)
+
+        // test.blog + wp.com
+        XCTAssertEqual(1, accountService.numberOfAccounts())
+        // dotcom1.wordpress.com + jetpack.example.com
+        XCTAssertEqual(2, wpComAccount.blogs.count)
+        // test.blog + wp.com + jetpack (wpcc)
+        XCTAssertEqual(3, Blog.count(in: mainContext))
+
+        let testBlog = wpComAccount.blogs.first { $0.dotComID?.intValue == 1 }
+        XCTAssertNotNil(testBlog)
+        XCTAssertEqual(testBlog?.xmlrpc, "https://dotcom1.wordpress.com/xmlrpc.php")
+
+        let testBlog2 = wpComAccount.blogs.first { $0.dotComID?.intValue == 2 }
+        XCTAssertNotNil(testBlog2)
+        XCTAssertEqual(testBlog2?.xmlrpc, "http://jetpack.example.com/xmlrpc.php")
+    }
+
+    func testSyncBlogsMigratesJetpackSSL() {
+        HTTPStubs.stubRequest(forEndpoint: "me/sites",
+                              withFileAtPath: OHPathForFile("me-sites-with-jetpack.json", Self.self)!)
+        let saveExpectation = expectation(forNotification: .NSManagedObjectContextDidSave, object: mainContext)
+
+        let wpComAccount = accountService.createOrUpdateAccount(withUsername: "user", authToken: "token")
+        let dotcomBlog = Blog.createBlankBlog(with: wpComAccount)
+        dotcomBlog.xmlrpc = "https://dotcom1.wordpress.com/xmlrpc.php"
+        dotcomBlog.url = "https://dotcom1.wordpress.com/"
+        dotcomBlog.dotComID = NSNumber(integerLiteral: 1)
+
+        let jetpackBlog = Blog.createBlankBlog(in: mainContext)
+        jetpackBlog.username = "jetpack"
+        jetpackBlog.xmlrpc = "https://jetpack.example.com/xmlrpc.php" // now in https
+        jetpackBlog.url = "https://jetpack.example.com/" // now in https
+
+        // wait on the merge to be completed
+        wait(for: [saveExpectation], timeout: timeout)
+
+        XCTAssertEqual(1, accountService.numberOfAccounts())
+        // test.blog + wp.com + jetpack (legacy)
+        XCTAssertEqual(3, Blog.count(in: mainContext))
+        // dotcom1.wordpress.com
+        XCTAssertEqual(1, wpComAccount.blogs.count)
+
+        let syncExpectation = expectation(description: "Blogs sync")
+        blogService.syncBlogs(for: wpComAccount) {
+            syncExpectation.fulfill()
+        } failure: { _ in
+            XCTFail("Sync blogs shouldn't fail")
+        }
+        wait(for: [syncExpectation], timeout: 5.0)
+
+        // test.blog + wp.com
+        XCTAssertEqual(1, accountService.numberOfAccounts())
+        // dotcom1.wordpress.com + jetpack.example.com
+        XCTAssertEqual(2, wpComAccount.blogs.count)
+        // test.blog + wp.com + jetpack (wpcc)
+        XCTAssertEqual(3, Blog.count(in: mainContext))
+    }
+
+    // MARK: Jetpack Individual Plugins
+
+    func test_jetpackIsConnectedWithoutFullPlugin_givenIndividualPluginOnly_returnsTrue() {
```

Comment:
Nitpick: there's no underscore after the test on the functions above :D Let's keep them uniform?

Reply:
Good catch. These were previously from a separate Swift extension file for the unit test, but then I decided to migrate the legacy one and join them together 😅 . This is now addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20031/commits/22b1d9d4a4ad47826cf6cf522e9c7948f3c4da41.